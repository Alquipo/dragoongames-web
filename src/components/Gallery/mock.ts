export default [
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 1'
  },
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 2'
  },
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 3'
  },
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 4'
  },
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 5'
  },
  {
    src: 'https://source.unsplash.com/collection/4757197/1042x580',
    label: 'Gallery Image 6'
  }
]
