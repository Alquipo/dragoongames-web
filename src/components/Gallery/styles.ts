import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    .slick-prev,
    .slick-next {
      display: block;
      color: ${theme.colors.white};
      cursor: pointer;
      position: absolute;
      top: 50%;
      width: 2.5rem;
      height: 2.5rem;
      padding: 0;
      transform: translate(0, -50%);
    }

    .slick-prev {
      left: -${theme.spacings.xxlarge};
    }

    .slick-next {
      right: -${theme.spacings.xxlarge};
    }

    .slick-prev.slick-disabled,
    .slick-next.slick-disabled {
      visibility: hidden;
    }

    .slick-slide > div {
      margin: 0 ${theme.spacings.xsmall};
      cursor: pointer;
    }

    .slick-list {
      margin: 0 -${theme.spacings.xsmall};
    }

    ${media.lessThan('huge')`
      overflow-x: hidden;
    `}
  `}
`
