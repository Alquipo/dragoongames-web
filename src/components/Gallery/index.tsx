import { ArrowBackIos as ArrowLeft } from '@styled-icons/material-outlined/ArrowBackIos'
import { ArrowForwardIos as ArrowRight } from '@styled-icons/material-outlined/ArrowForwardIos'

import Slider, { SliderSettings } from 'components/Slider'

import * as S from './styles'

const settings: SliderSettings = {
  arrows: true,
  slidesToShow: 4,
  infinite: false,
  responsive: [
    {
      breakpoint: 1375,
      settings: {
        arrows: false,
        slidesToShow: 3.2,
        draggable: true
      }
    },
    {
      breakpoint: 1024,
      settings: {
        arrows: false,
        slidesToShow: 2.2,
        draggable: true
      }
    },
    {
      breakpoint: 768,
      settings: {
        arrows: false,
        slidesToShow: 2.2,
        draggable: true
      }
    }
  ],
  nextArrow: <ArrowRight aria-label="next image" />,
  prevArrow: <ArrowLeft aria-label="previous image" />
}

export type GalleryImageProps = {
  src: string
  label: string
}

export type GalleryProps = {
  items: GalleryImageProps[]
}

const Gallery = ({ items }: GalleryProps) => (
  <S.Wrapper>
    <Slider settings={settings}>
      {items.map((item, index) => (
        <img
          key={`thumb-${index}`}
          role="button"
          src={item.src}
          alt={`Thumb - ${item.label}`}
        />
      ))}
    </Slider>
  </S.Wrapper>
)

export default Gallery
