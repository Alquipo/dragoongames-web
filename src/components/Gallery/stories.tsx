import { Story, Meta } from '@storybook/react/types-6-0'
import Gallery, { GalleryProps } from '.'
import items from './mock'

export default {
  title: 'Game/Gallery',
  component: Gallery,
  args: {
    items
  },
  parameters: {
    backgrounds: {
      default: 'dragoon-dark'
    }
  }
} as Meta

export const Default: Story<GalleryProps> = (args) => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <Gallery {...args} />
  </div>
)
