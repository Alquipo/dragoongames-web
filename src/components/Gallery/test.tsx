// import { screen } from '@testing-library/react'

import 'match-media-mock'
import { renderWithTheme } from 'utils/tests/helpers'
import Gallery from '.'

const items = [
  {
    src: '/img/games/cyberpunk-1.jpg',
    label: 'Gallery Image 1'
  },
  {
    src: '/img/games/cyberpunk-2.jpg',
    label: 'Gallery Image 2'
  },
  {
    src: '/img/games/cyberpunk-3.jpg',
    label: 'Gallery Image 3'
  },
  {
    src: '/img/games/cyberpunk-4.jpg',
    label: 'Gallery Image 4'
  },
  {
    src: '/img/games/cyberpunk-5.jpg',
    label: 'Gallery Image 5'
  },
  {
    src: '/img/games/cyberpunk-6.jpg',
    label: 'Gallery Image 6'
  }
]

describe('<Gallery />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Gallery items={items} />)

    // expect(
    //   screen.getByRole('heading', { name: /Gallery/i })
    // ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
