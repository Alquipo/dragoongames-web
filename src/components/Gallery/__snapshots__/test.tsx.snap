// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> should render the heading 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c1 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c1 .slick-list:focus {
  outline: none;
}

.c1 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c1 .slick-slider .slick-track,
.c1 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c1 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c1 .slick-track:before,
.c1 .slick-track:after {
  display: table;
  content: '';
}

.c1 .slick-track:after {
  clear: both;
}

.c1 .slick-loading .slick-track {
  visibility: hidden;
}

.c1 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c1 [dir='rtl'] .slick-slide {
  float: right;
}

.c1 .slick-slide img {
  display: block;
}

.c1 .slick-slide.slick-loading img {
  display: none;
}

.c1 .slick-slide.dragging img {
  pointer-events: none;
}

.c1 .slick-initialized .slick-slide {
  display: block;
}

.c1 .slick-loading .slick-slide {
  visibility: hidden;
}

.c1 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c1 .slick-arrow.slick-hidden {
  display: none;
}

.c0 .slick-prev,
.c0 .slick-next {
  display: block;
  color: #FAFAFA;
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c0 .slick-prev {
  left: -5.6rem;
}

.c0 .slick-next {
  right: -5.6rem;
}

.c0 .slick-prev.slick-disabled,
.c0 .slick-next.slick-disabled {
  visibility: hidden;
}

.c0 .slick-slide > div {
  margin: 0 1.6rem;
  cursor: pointer;
}

.c0 .slick-list {
  margin: 0 -1.6rem;
}

@media (max-width:1440px) {
  .c0 {
    overflow-x: hidden;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <svg
        aria-hidden="true"
        aria-label="previous image"
        class="c2 slick-arrow slick-prev slick-disabled"
        data-role="none"
        fill="currentColor"
        focusable="false"
        style="display: block;"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
          opacity="0.87"
        />
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 1"
                role="button"
                src="/img/games/cyberpunk-1.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 2"
                role="button"
                src="/img/games/cyberpunk-2.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 3"
                role="button"
                src="/img/games/cyberpunk-3.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 4"
                role="button"
                src="/img/games/cyberpunk-4.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="4"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 5"
                role="button"
                src="/img/games/cyberpunk-5.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="5"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Thumb - Gallery Image 6"
                role="button"
                src="/img/games/cyberpunk-6.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
      <svg
        aria-hidden="true"
        aria-label="next image"
        class="c2 slick-arrow slick-next"
        data-role="none"
        fill="currentColor"
        focusable="false"
        style="display: block;"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 24H0V0h24v24z"
          fill="none"
          opacity="0.87"
        />
        <path
          d="M6.49 20.13l1.77 1.77 9.9-9.9-9.9-9.9-1.77 1.77L14.62 12l-8.13 8.13z"
        />
      </svg>
    </div>
  </section>
</div>
`
